### ISSUE TITLE
Connect: RBAC won't Filter By Connector Name

### ISSUE THREAD

**Issue by schaubce** [provectus/kafka-ui#4253]
### Issue submitter TODO list

- [X] I've looked up my issue in [FAQ](https://docs.kafka-ui.provectus.io/faq/common-problems)
- [X] I've searched for an already existing issues [here](https://github.com/provectus/kafka-ui/issues)
- [X] I've tried running `master`-labeled docker image and the issue still persists there
- [X] I'm running a supported version of the application which is listed [here](https://github.com/provectus/kafka-ui/blob/master/SECURITY.md)

### Describe the bug (actual behavior)

We have the following configuration (I have removed other resources topic/cluster/etc. for simplicity)
`  - name: <adGroupName>
      clusters:
      - primary-inf-grmus-kafka-np
      - primary-inf-hs-kafka-np
      subjects: []
      permissions: 
      - resource: connect
        value: ibm-mq-src--mqconnector--v1
        actions:
        - VIEW
        - RESTART
        - EDIT`

When we refresh configs within Kafka-UI, no connectors are shown.  If there is no specific filter (.*), it works appropriately, and shows all connectors. 
![Screenshot 2023-09-20 at 9 40 46 AM](https://github.com/provectus/kafka-ui/assets/6474582/2db09bb2-1086-4278-91d5-023dfb53a23f)

### Expected behavior

Similar to topics configuration (which we have working) , we'd expect to be able to use regex to filter out different connector names to limit who can view specific connectors.  We did see this update in the documentation, but unsure if it is related, as the other config examples show it should work with regex: 
`- resource: connect
          value: "local"
          actions: [ view, edit, create ]
        # connectors selector not implemented yet, use connects
        #      selector:
        #        connector:
        #          name: ".*"
        #          class: 'com.provectus.connectorName'`

vs 

`- resource: connect
          value: ".*"
          actions: [ view ]`

### Your installation details

Hash: 56fa824 (Version 0.7.1)

### Steps to reproduce

Can use normal set up for topics, cluster etc. Just need to limit the connect resource "value" to a single connector. 
`  - name: <adGroupName>
      clusters:
      - primary-inf-grmus-kafka-np
      - primary-inf-hs-kafka-np
      subjects: []
      permissions: 
      - resource: connect
        value: ibm-mq-src--mqconnector--v1
        actions:
        - VIEW
        - RESTART
        - EDIT`

### Additional context

- We have tested different connector types to see if that impacted anything 
- We have tested different connector name regex: full name, prefix

**Comment by tufank** [comment]
Hello,

We are also having the same issue, and we are not sure what 'use connects' means in the comment 'selector not implemented yet...'., which also makes the meaning of 'value' in 'resource: connect' ambiguous. 

Is there currently a way or a workaround to be able to define a regex for connector names? Or any documentation that makes it more clear how to use 'resource: connector' when we want to limit access to connectors?

Thanks in advance for any help.

**Comment by Haarolean** [comment]
Just to clarify this for everyone here,

there are two things you have: **connects** (kafka-connect node) and **connectors** (data apps for KC within a connect).

With a setup like this:
```
kafka:
  clusters:
    - name: local
      # ...
      kafkaConnect:
        - name: first
          address: http://localhost:8083
        - name: second
          address: http://localhost:8084
```

You would see something like this:
<img width="632" alt="Screenshot 2024-02-01 at 20 13 20" src="https://github.com/provectus/kafka-ui/assets/1494347/49f2dc4e-6784-4954-a9b9-dbaae07e5d7c">

And if you limit the access with RBAC allowing only "first" **connect** access it would look like this:
<img width="580" alt="Screenshot 2024-02-01 at 20 14 08" src="https://github.com/provectus/kafka-ui/assets/1494347/b49347b8-a7db-43ce-bd13-e2f435922309">

So this works as intended. And the commented-out part of the config suggests that limiting access to specific connectors within a connect is not currently implemented, so, you have to limit the access to the whole connect.

**Comment by tufank** [comment]
Thank you, I appreciate the clarification. It is now clear that the 'value' field of the 'resource: connect' entry in RBAC definitions corresponds to the name fields of clusters.kafkaConnect entries, and currently it is not possible to limit permissions by the name or regex of the connectors defined within the Connect cluster itself.

**Comment by Haarolean** [comment]
@tufank yeah, that's right. If you're looking for such a feature, I invite you to raise an issue here: https://github.com/kafbat/kafka-ui as this repo is not maintained anymore (#4255)
